contracts/Ethleverage.sol

```diff
@@ -0,0 +1,292 @@
+import "./Interface/IpethContract.sol";
+import "./Interface/IMKRContract.sol";
+import "./Interface/IMoneyMaker.sol";
+import "./Interface/ILiquidator.sol";
+
+
+//To-Do: be sure to complete the ETH_Address authorization
+contract Ethleverage {
+  using SafeMath for uint;
+
+  //events
+
+  event CDPOpened(bytes32 ID);
+  event PethLocked(bytes32 ID, uint recycledPeth);
+  event dai2Peth(uint recycledPeth);
+
+  event LogCDPAddressChanged(address oldAddress, address newAddress);
+  event LogDaiAddressChanged(address oldAddress, address newAddress);
+
+  //Variables
+  struct Investor {
+    uint layers;        // number of layers down
+    uint prinContr;     // principle contribution in Eth
+    uint CR;            // collatorization ratio
+    bytes32 cdps;     //array of CDPs (only one!)
+    uint daiAmountFinal;  //amount of Dai that an investor has after leveraging
+    uint ethAmountFinal; //resulting amount of ETH that an investor gets after liquidating
+  }
+
+  mapping (address => Investor) public investors;
+  address[] public investorAddresses;
+
+  address public owner;
+
+  address public CDPContract;
+  address public DaiContract;
+  address public wethContract;
+  address public pethContract;
+  address public MKRContract;
+	address public moneyMakerKovan = 0xd87856163f409777df41DDFBF37A66369E028FA9;
+  address public liquidKovan =  0xc936749D2D0139174EE0271bD28325074fdBC654; //https://github.com/makerdao/sai/blob/4b0c94b8ef2d8e0951dd0a0eee7c0fce5f5dbb49/src/tap.sol
+  address public unknownContract = 0xc936749D2D0139174EE0271bD28325074fdBC654; //https://kovan.etherscan.io/address/0xc936749d2d0139174ee0271bd28325074fdbc654
+
+  uint public makerLR;
+  uint public ethCap = 10000;
+  uint public layers = 3;
+
+
+  //Modifiers
+  modifier onlyOwner {
+    require(msg.sender == owner);
+      _;
+  }
+
+  modifier onlyInvestor {
+    require(investors[msg.sender].prinContr != 0);
+    _;
+  }
+
+  //Functions
+  function Ethleverage(
+    address _CDPaddr,
+    address _Daiaddr,
+    address _wethaddr,
+    address _pethaddr,
+    address _mkrContract,
+    uint _liquidationRatio
+  ) public {
+    owner = msg.sender;
+    CDPContract = _CDPaddr;
+    DaiContract = _Daiaddr;
+    wethContract = _wethaddr;
+    pethContract = _pethaddr;
+    MKRContract = _mkrContract;
+    makerLR = _liquidationRatio;
+  }
+
+	function initialize() public {
+      //for some reason the variables ARE initialized in the constructor function, but an error is thrown without reassigning them here.
+      owner = msg.sender;
+      moneyMakerKovan = 0xd87856163f409777df41DDFBF37A66369E028FA9;
+      liquidKovan =  0xc936749D2D0139174EE0271bD28325074fdBC654; //https://github.com/makerdao/sai/blob/4b0c94b8ef2d8e0951dd0a0eee7c0fce5f5dbb49/src/tap.sol
+      unknownContract = 0xc936749D2D0139174EE0271bD28325074fdBC654; //https://kovan.etherscan.io/address/0xc936749d2d0139174ee0271bd28325074fdbc654
+      makerLR = 151;
+      ethCap = 10000;
+      layers = 3;
```

Comment:
This was also assigned globally earlier in the code, let's get rid of one of them